tasks.php

```diff
@@ -1169,15 +1169,15 @@ function TaskForm($task)
     global $task_assignees_array;
     global $priority_array, $tasks_url;
 
+    // Don't want to permit setting status to "Closed" or "Reopened" when creating/editing a task
+    unset($tasks_status_array[get_property_key("Closed", $tasks_status_array)]);
+    unset($tasks_status_array[get_property_key("Reopened", $tasks_status_array)]);
+
     // Non-managers can only set the task status to New.
     if (!user_is_a_sitemanager() && !user_is_taskcenter_mgr()) {
         $tasks_status_array = [get_property_key("New", $tasks_status_array) => "New"];
     }
```

Comment:
Can you elaborate on how this change resolves the problem? This redefines the array for ordinary users so it shouldn't matter if we unset fields above or below the section. I'm very likely missing some nuance which is a code-smell.

Reply:
The problem is the get_property_key(). If the array is redifined to be just "New" then the values "Closed" or "Reopened" no longer exist and the function get_property_key() throws an error.
Perhaps a better fix would be to say
```
    // Non-managers can only set the task status to New.
    if (!user_is_a_sitemanager() && !user_is_taskcenter_mgr()) {
        $tasks_status_array = [get_property_key("New", $tasks_status_array) => "New"];
    } else {
      // Don't want to permit setting status to "Closed" or "Reopened" when creating/editing a task
      unset($tasks_status_array[get_property_key("Closed", $tasks_status_array)]);
      unset($tasks_status_array[get_property_key("Reopened", $tasks_status_array)]);
}
```

Reply:
Ah, got it. I do like the if/else block better 👍

Reply:
I'll make it so.